apps/confidential/confidential_121_plus.go

```diff
@@ -0,0 +1,21 @@
+//go:build go1.21
+
+package confidential
+
+import (
+	"fmt"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/logger"
+)
+
+//  WithLogger allows for a custom logger to be set.
+func WithLogger(l interface{}) Option {
```

Comment:
`interface{}` effectively disables compiler type checking here, making this function difficult to use and turning a class of build errors into runtime errors--not great. This option only works if `l` is a `*slog.Logger`; should that be the parameter's type?